bundles/org.openhab.binding.mail/README.md

```diff
@@ -145,7 +145,7 @@ Examples:
 
 ```java
 val mailActions = getActions("mail","mail:smtp:samplesmtp")
-val success = mailActions.sendMail("mail@example.com", "Test subject", "This is the mail content.")
+var success = mailActions.sendMail("mail@example.com", "Test subject", "This is the mail content.")
```

Comment:
@andrewfg - example of what I mean:
```suggestion
var success = mailActions.sendMail("mail@example.com", "Test subject", "This is the mail content.");
```

Reply:
I am not convinced that such semi colons are actually good practice. Certainly not in DSL  and maybe not in Javascript. Does OH have an official style guideline? Anyway I do not intend to change those things on the two lines that I edited. It would open the can or worms of editing all lines in this file, and indeed all lines in all files. Where would one stop? @rkoshak specifically asked to change val to var because that dies have a meaningful functional impact (see forum post) and any other change neither has an impact on that issue, nor IMHO brings any other concrete benefit..

Reply:
> I am not convinced that such semi colons are actually good practice. Certainly not in DSL and maybe not in Javascript.
Be very careful to draw conclusions from one language (Rules DSL) and applying them to another (JS). Conventions like this are usually there for a reason in each language and they are not the same across all languages. Since this is a JS example it should follow JS conventions, and all JS examples in this file should use the same conventions. 

> It would open the can or worms of editing all lines in this file, and indeed all lines in all files. Where would one stop?
Just this one file and it would only impact the JS examples in this one file. Since there are only two JS examples in this file, it would only apply to those two.

There are a few cases where a semicolon is required. But for the most part the semicolon is optional. It's a surprisingly complicated topic really. It's technically required by ECMASScript but some parsers will automatically insert the semicolon for you while others (and the spec allows) the lack of a semicolon to be ignored by the parser instead of inserting one for you. Looking at a few style guides I see no consensus. 
- Google says to always use a semicolon with some exceptions
- W3 Schools says to always use a semicolon on simple statements, not complex statements (e.g. if, for, while)
- MDN doesn't state a rule but it's examples appear to follow W3 Schools
- StandardJS says to never use semicolons except where it's absolutely required
- Node.js says to always use them

🤷 

There are a few cases where not adding a semicolon can indeed cause problems but those are rare and unlikely to come up in typical rules development. 

I tend to use them but don't have a strong opinion on whether we should adopt a coding standard and what that standard should be. `openhab-js` seems to follow a coding style that uses them for what it's worth and since JS Scripting provides a Node.js like environment that style guide should carry some weight. 

Personally all I care about are that the examples are correct (i.e. will run as written) and that all the examples on any individual page use a consistent style. But don't worry about adopting a standard on this one page exploding. Just worry about this one page.

Reply:
Closing this thread, as I will not implement it.